datashape/tests/test_coretypes.py

```diff
@@ -315,3 +316,15 @@ def test_duplicate_field_names_fails():
                           'var * {"func-y": (A) -> var * {a: 10 * float64}}'])
 def test_repr_of_eval_is_dshape(ds):
     assert eval(repr(dshape(ds))) == dshape(ds)
+
+
+@pytest.mark.parametrize(
+    'ds',
+    [('int32', int32),
+     ('int64', int64),
+     ('float32', float32),
+     ('float64', float64),
+     ('string[10, "A"]', String(10, 'A'))],
+)
+def test_range(ds):
+    assert dshape('range[{d}]'.format(d=ds[0])) == dshape(str(Range(ds[1])))
```

Comment:
I guess I was trying to test a subtype with a parameter. Could someone not provide a range of say, column names that would be expressed as range[string]?